polaris-react/src/components/Page/Page.stories.tsx

```diff
@@ -315,3 +327,96 @@ export function WithDivider() {
     </Page>
   );
 }
+
+export function WithCustomPopover() {
+  const [popoverVisible, setPopoverVisible] = useState(false);
+  const [bannerVisible, setBannerVisible] = useState(false);
+  const [popoverPosition, setPopoverPosition] = useState({top: 0, right: 0});
+  const {mdUp} = useBreakpoints();
+
+  const activator = <div style={{visibility: 'hidden'}}>I'm a button</div>;
+
+  const customAboutAction = {
+    content: 'About',
+    accessibilityLabel: 'Secondary action label',
+    onAction: (ref: React.RefObject<HTMLElement>) => {
+      if (!ref.current) {
+        return;
+      }
+
+      const {right, top, height} = ref.current.getBoundingClientRect();
+      setPopoverPosition({
+        top: top + height,
+        right: window.innerWidth - right,
+      });
+
+      if (mdUp) {
+        setPopoverVisible(!popoverVisible);
+      } else {
+        setBannerVisible(!bannerVisible);
+      }
+    },
```

Comment:
I know this is just an example, but the more and more I play around with it in Storybook, the more it feels like consumers would need to add a lot of code here to make things work well. For example, the popover doesn't appear until there's a page resize event (I assume because `ref.current` is falsy on first render), responding to resize events, etc.

I wonder if a better approach is to find a way to past a ref into the `<Popover />` props for `activator` so that all these things are handled for us.